gio/src/auto/task.rs

```diff
@@ -0,0 +1,243 @@
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_44")))]
+use glib::signal::connect_raw;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_44")))]
+use glib::signal::SignalHandlerId;
+use glib::translate::*;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_44")))]
+use std::boxed::Box as Box_;
+use std::fmt;
+#[cfg(any(feature = "v2_44", feature = "dox"))]
+#[cfg_attr(feature = "dox", doc(cfg(feature = "v2_44")))]
+use std::mem::transmute;
+use std::ptr;
+
+glib::wrapper! {
+    pub struct Task(Object<ffi::GTask, ffi::GTaskClass>) @implements AsyncResult;
+
+    match fn {
+        get_type => || ffi::g_task_get_type(),
+    }
+}
+
+impl Task {
+    //#[doc(alias = "g_task_new")]
+    //pub fn new<P: IsA<Cancellable>, Q: FnOnce(Result<(), glib::Error>) + 'static>(source_object: Option<&glib::Object>, cancellable: Option<&P>, callback: Q) -> Task {
+    //    unsafe { TODO: call ffi:g_task_new() }
+    //}
+
+    //#[doc(alias = "g_task_attach_source")]
+    //pub fn attach_source<P: Fn() -> bool + 'static>(&self, source: &glib::Source, callback: P) {
+    //    unsafe { TODO: call ffi:g_task_attach_source() }
+    //}
+
+    #[doc(alias = "g_task_get_cancellable")]
+    pub fn get_cancellable(&self) -> Option<Cancellable> {
+        unsafe { from_glib_none(ffi::g_task_get_cancellable(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_get_check_cancellable")]
+    pub fn get_check_cancellable(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_get_check_cancellable(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_44", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_44")))]
+    #[doc(alias = "g_task_get_completed")]
+    pub fn get_completed(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_get_completed(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_get_context")]
+    pub fn get_context(&self) -> Option<glib::MainContext> {
+        unsafe { from_glib_none(ffi::g_task_get_context(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(feature = "v2_60", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_60")))]
+    #[doc(alias = "g_task_get_name")]
+    pub fn get_name(&self) -> Option<glib::GString> {
+        unsafe { from_glib_none(ffi::g_task_get_name(self.to_glib_none().0)) }
+    }
+
+    #[doc(alias = "g_task_get_return_on_cancel")]
+    pub fn get_return_on_cancel(&self) -> bool {
+        unsafe { from_glib(ffi::g_task_get_return_on_cancel(self.to_glib_none().0)) }
+    }
+
+    //#[doc(alias = "g_task_get_source_tag")]
+    //pub fn get_source_tag(&self) -> /*Unimplemented*/Option<Fundamental: Pointer> {
+    //    unsafe { TODO: call ffi:g_task_get_source_tag() }
+    //}
+
+    //#[doc(alias = "g_task_get_task_data")]
+    //pub fn get_task_data(&self) -> /*Unimplemented*/Option<Fundamental: Pointer> {
```

Comment:
Same question for this: a `GTask` subclass would be used here in case of bindings to pass around custom data?

Reply:
In this case probably an option would be to subclass `Task` indeed